akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala

```diff
@@ -327,6 +329,8 @@ abstract class MessageDispatcherConfigurator(_config: Config, val prerequisites:
     def configurator(executor: String): ExecutorServiceConfigurator = executor match {
       case null | "" | "fork-join-executor" ⇒ new ForkJoinExecutorConfigurator(config.getConfig("fork-join-executor"), prerequisites)
       case "thread-pool-executor"           ⇒ new ThreadPoolExecutorConfigurator(config.getConfig("thread-pool-executor"), prerequisites)
+      case "affinity-pool-executor"         ⇒ new AffinityPoolConfigurator(config.getConfig("affinity-pool-executor"), prerequisites)
+
       case fqcn ⇒
```

Comment:
It must be a separate module, because of the dependency.
I'm thinking if we should have that module cover a bit more than just the affinity. Like some "high performance stuff", e.g. (allocation free) mailboxes and stuff.
Naming is difficult, as usual.